### PR TITLE
Initial commit of Dockerization of NVD CVE conversion

### PR DESCRIPTION
**Pull request by andrewpollock**

### CODE REVIEW

File: deployment/clouddeploy/gke-workers/base/nvd-cve-osv.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  schedule: "0 6 * * *"
+  concurrencyPolicy: Forbid
+  jobTemplate:
+    spec:
+      activeDeadlineSeconds: 86400
+      template:
+        spec:
+          containers:
+          - name: nvd-cve-osv
+            image: nvd-cve-osv
```
Comment (michaelkedar) [inline deployment/clouddeploy/gke-workers/base/nvd-cve-osv.yaml:16]: Going to need to build this image in Cloud Build like here:
https://github.com/google/osv.dev/blob/5433f3c427dd76c4618c084389107ac6140ab0c2/deployment/build-and-stage.yaml#L151-L159

Comment (michaelkedar) [inline deployment/clouddeploy/gke-workers/base/nvd-cve-osv.yaml:16]: And also need to add the image name substitution here: https://github.com/google/osv.dev/blob/5433f3c427dd76c4618c084389107ac6140ab0c2/deployment/build-and-stage.yaml#L181-L194

File: vulnfeeds/cpp/build.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
```
Comment (michaelkedar) [inline vulnfeeds/cpp/build.sh:1]: nit: Is this script only used for building the image manually and do we actually need it?

File: deployment/clouddeploy/gke-workers/environments/oss-vdb-test/kustomization.yaml [hunk 3]
```diff
@@ -11,3 +11,4 @@ patches:
 - debian-first-version.yaml
 - debian-copyright-mirror.yaml
 - gen-cperepos-map.yaml
+- nvd-cve-osv.yaml
```
[the remark this answers is not in the document]
Comment (andrewpollock) [reply deployment/clouddeploy/gke-workers/environments/oss-vdb-test/kustomization.yaml:14]: Yeah that was intentional, I want to validate everything in staging first.

File: deployment/clouddeploy/gke-workers/base/nvd-cve-osv.yaml (hunk 1 above)
Comment (andrewpollock) [reply deployment/clouddeploy/gke-workers/base/nvd-cve-osv.yaml:16]: Thanks, added in [90d0746](https://github.com/google/osv.dev/pull/1306/commits/90d07468737e2b1c4963f0fff12dfec3288c063a)

Comment (andrewpollock) [reply deployment/clouddeploy/gke-workers/base/nvd-cve-osv.yaml:16]: Thanks, added in [90d0746](https://github.com/google/osv.dev/pull/1306/commits/90d07468737e2b1c4963f0fff12dfec3288c063a)

File: vulnfeeds/cpp/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+COPY go.mod go.sum ./
+RUN go mod download && go mod verify
+
+COPY . .
+RUN CGO_ENABLED=0 go build -v -o /usr/local/bin ./cpp ./cmd/download-cves
+
+FROM gcr.io/google.com/cloudsdktool/google-cloud-cli:alpine AS runtime
+
+COPY --from=buildenv /usr/local/bin ./usr/local/bin
```
[the remark this answers is not in the document]
Comment (andrewpollock) [reply vulnfeeds/cpp/Dockerfile:27]: It's empty prior to my build artifacts landing there.

File: vulnfeeds/cpp/build.sh (hunk 2 above)
Comment (andrewpollock) [reply vulnfeeds/cpp/build.sh:1]: There's no other good breadcrumb trail near the code/Dockerfile for what the actual image name is.

If we can come up with some other convention to help the unfamiliar figure out the name of the Docker image that gets built, I'd be fine with delousing this.

Comment (michaelkedar) [reply vulnfeeds/cpp/build.sh:1]: Yeah, makes sense - happy to keep it.
In the future, I'd like to think of some other convention - osv.dev having a bunch of unused build scripts can be a bit confusing sometimes.

File: vulnfeeds/cpp/build.sh [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+set -ex
+
+cd ../..
```
Comment (michaelkedar) [inline vulnfeeds/cpp/build.sh:20]: I think this should just be `cd ..` if I'm following the directory structure right.

**Review by michaelkedar — APPROVED** [review]

File: vulnfeeds/cpp/build.sh (hunk 5 above)
Comment (andrewpollock) [reply vulnfeeds/cpp/build.sh]: Good catch, thank you. Corrected in [c1dcd02](https://github.com/google/osv.dev/pull/1306/commits/c1dcd02d5aeeaedfbe9a47719bdcf0517287064e)

**Pull request merged**